File: Los Tablones de la Valla/Los Tablones de la Valla.cpp
#include <stdio.h>
#define Hd main(void)
Hd
{
    int n, k;
    scanf("%d %d", &n, &k);
    int alturas[n];

    for (int i = 0; i < n; i++) {
        scanf("%d", &alturas[i]);
    }

    int actsum = 0;
    int minsum = 0;
    int minind = 0;
    
    for (int i = 0; i < k; i++) {
        actsum += alturas[i];
    }
    
    minsum = actsum;

    for (int i = 1; i < n - k + 1; i++) {
        actsum = actsum - alturas[i - 1] + alturas[i + k - 1];
        if (actsum < minsum) {
            minsum = actsum;
            minind = i;
        }
    }

    printf("%d", minind + 1);
}

File: Los Tablones de la Valla/Los Tablones de la Valla.py
#define Hd 
def minh(n, k, alturas):
    actsum = sum(alturas[:k])
    minsum = actsum
    minind = 0

    for i in range(1, n - k + 1):
        actsum = actsum - alturas[i - 1] + alturas[i + k - 1]
        if actsum < minsum:
            minsum = actsum
            minind = i

    return minind

n, k = map(int, input().split())
alturas = list(map(int, input().split()))

minind = minh(n, k, alturas)

print(minind + 1)